File: 프로그래머스/Level1/예산.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> d, int budget) {
    int answer = 0;
    int result = 0;
    
    sort(d.begin(),d.end(), less<int>());
    
    for(int num : d){
        result += num;
        if(result > budget) break;
        else answer ++;
    }
    return answer;
}

File: 프로그래머스/Level1/예산.js
function solution(d, budget) {
    let answer = 0, tmp = 0;
    
    d = d.sort((a,b) => a-b);
    for(let i=answer; i<d.length; i++){
        if(tmp + d[answer] > budget) break;
        tmp += d[answer++];
    }
    
    return answer;
}
